src/constraint/NonColliding.cpp

```diff
@@ -65,11 +65,48 @@ void NonColliding::addPairwiseCheck(
 }
 
 //==============================================================================
+void NonColliding::removePairwiseCheck(
+    std::shared_ptr<dart::collision::CollisionGroup> _group1,
+    std::shared_ptr<dart::collision::CollisionGroup> _group2)
+{
+  auto it = std::find(
+      groupsToPairwiseCheck.begin(),
+      groupsToPairwiseCheck.end(),
+      std::make_pair(_group1, _group2));
+  if (it != groupsToPairwiseCheck.end())
+  {
+    groupsToPairwiseCheck.erase(it);
+    return;
+  }
+
+  // Check for reverse pair
```

Comment:
I think we could remove this double checking by storing the groups in a pair (e.g., ascending or descending in the pointer addresses). For this, we should:
* change `addPairwiseCheck(...)` to add group pair in ascending order something like:
```cpp
  if (_group1 < _group2)
    groupsToPairwiseCheck.emplace_back(std::move(_group1), std::move(_group2));
  else
    groupsToPairwiseCheck.emplace_back(std::move(_group2), std::move(_group1));
```
* change `removePairwiseCheck(...)` to find the ascending-ordered group pair

More importantly, I think we could use `std::unordered_set` instead of `std::vector` for better performance in adding and removing because we don't need random access, and the order of pairs is not important. In case we switch to `std::unordered_set` we might directly use `std::unordered_set::erase` without finding the pair.